src/collections/delivery/operations/postal-code-lookup.operation.ts

```
@@ -44,6 +44,9 @@ export class PostalCodeLookupOperation implements Operation {
     if (!verifyPostalCodeLookupSpec(postalCodeLookupSpec)) {
       throw new BlError(`Malformed PostalCodeSpec`).code(701);
     }
+    if (postalCodeLookupSpec.postalCode.length !== 4) {
+      return new BlapiResponse([{ postalCity: null }]);
+    }
```

I've changed my mind a little, I think we should return a bad request if the postal code does not fit the `[0-9]{4}`-regex. The frontend can know that for itself instead of sending unnecessary requests. I therefore think this check should also check that they're all digits, and throw instead of returning null. However, we should still return null if the postal code is not assigned.

---

Sure, moved the logic to the verifySpec